### ISSUE TITLE
Wrong transitive dependency resolution when mixing maven dep.managment with property overrides

### ISSUE THREAD

**Issue by tmohme** [gradle/gradle#1173]
Under certain conditions Gradle 3.3 resolves transitive dependencies to wrong versions.  
This seems to be related a combination of 
- version-property declaration & dependency-managent in "grand-parent" POM
- version-property override in parent POM
- actual dependency in child POM

### Expected Behavior
Maven honors the overridden version-property when resolving the actual dependency, thus resolving to the version specified in the parent POM.
I expect the same from Gradle.

Example: 
`dependencies { compile 'de.agilecoders.wicket:wicket-bootstrap-extensions:jar:0.10.11'}`
Version 7.4.0 of org.apache.wicket:wicket-extensions:jar should be selected.

### Current Behavior
Currently, Gradle selects version 6.15.0 of org.apache.wicket:wicket-extensions

### Context
This lead to a non-working application because there are major changes between these versions.

### Steps to Reproduce (for bugs)
I created a [GitHub repo](https://github.com/tmohme/gradle-bug-demo) with both an `build.gradle` and a `pom.xml` for reproduction.

### Your Environment
 * Build scan URL: https://gradle.com/s/2aigh5fovdyl6

**Comment by bmuschko** [comment]
In the [parent POM that defines the dependency versions](https://search.maven.org/remotecontent?filepath=de/agilecoders/maven/maven-parent-pom/0.3.5/maven-parent-pom-0.3.5.pom) I see the following:

    <wicket.version>6.15.0</wicket.version>
    ...
    <dependency>
        <groupId>org.apache.wicket</groupId>
        <artifactId>wicket-extensions</artifactId>
        <version>${wicket.version}</version>
    </dependency>

Why do you think version 7.4.0 should be selected?

**Comment by tmohme** [comment]
You tracked it down to the *grand*-parent (de.agilecoders.maven:maven-parent-pom:0.3.5) of the original dependency.
Unfortunately the property `wicket.version` gets overridden in the [parent POM `de.agilecoders.wicket:bootstrap-parent:0.10.11`](https://repo1.maven.org/maven2/de/agilecoders/wicket/bootstrap-parent/0.10.11/bootstrap-parent-0.10.11.pom) in the middle of the hierarchy which does not a declare a dependency on `wicket-extensions` by itself.

**Comment by bmuschko** [comment]
OK, thx for the info. Could you verify that Maven resolves to version 7.4.0 for this use case?

**Comment by tmohme** [comment]
This is the output of `./mvnw dependency:tree` on my [demo-project](https://github.com/tmohme/gradle-bug-demo):
```[INFO] ------------------------------------------------------------------------
[INFO] Building bug-demo 1.0.0
[INFO] ------------------------------------------------------------------------
[INFO] 
[INFO] --- maven-dependency-plugin:2.8:tree (default-cli) @ bug-demo ---
[INFO] bug-demo:bug-demo:jar:1.0.0
[INFO] \- de.agilecoders.wicket:wicket-bootstrap-extensions:jar:0.10.11:compile
[INFO]    +- de.agilecoders.wicket:jquery-selectors:jar:0.2.3:compile
[INFO]    |  +- com.fasterxml.jackson.core:jackson-core:jar:2.6.0:compile
[INFO]    |  \- com.fasterxml.jackson.core:jackson-databind:jar:2.6.0:compile
[INFO]    |     \- com.fasterxml.jackson.core:jackson-annotations:jar:2.6.0:compile
[INFO]    +- de.agilecoders.wicket:wicket-bootstrap-core:jar:0.10.11:compile
[INFO]    |  +- de.agilecoders.wicket.webjars:wicket-webjars:jar:0.5.5:compile
[INFO]    |  +- org.webjars:bootstrap:jar:3.3.7-1:compile
[INFO]    |  +- org.webjars:modernizr:jar:2.8.3:compile
[INFO]    |  +- org.apache.wicket:wicket-core:jar:7.4.0:compile
[INFO]    |  |  \- org.apache.commons:commons-collections4:jar:4.1:compile
[INFO]    |  +- org.apache.wicket:wicket-request:jar:7.4.0:compile
[INFO]    |  \- org.slf4j:slf4j-api:jar:1.7.7:compile
[INFO]    +- org.apache.wicket:wicket-extensions:jar:7.4.0:compile
[INFO]    +- org.apache.wicket:wicket-util:jar:7.4.0:compile
[INFO]    |  +- commons-fileupload:commons-fileupload:jar:1.3.2:compile
[INFO]    |  \- commons-io:commons-io:jar:2.5:compile
[INFO]    +- org.webjars:Eonasdan-bootstrap-datetimepicker:jar:4.17.37-1:compile
[INFO]    +- org.webjars:momentjs:jar:2.14.1:compile
[INFO]    +- org.webjars:font-awesome:jar:4.6.3:compile
[INFO]    +- com.google.guava:guava:jar:19.0:compile
[INFO]    +- com.google.javascript:closure-compiler:jar:v20130603:compile
[INFO]    |  +- args4j:args4j:jar:2.0.16:compile
[INFO]    |  +- com.google.protobuf:protobuf-java:jar:2.4.1:compile
[INFO]    |  +- org.json:json:jar:20090211:compile
[INFO]    |  \- com.google.code.findbugs:jsr305:jar:1.3.9:compile
[INFO]    +- joda-time:joda-time:jar:2.3:compile
[INFO]    +- org.webjars:jquerypp:jar:1.0.1:compile
[INFO]    +- org.webjars:jquery-ui:jar:1.11.4:compile
[INFO]    |  \- org.webjars:jquery:jar:1.11.1:compile
[INFO]    +- org.webjars:typeaheadjs:jar:0.10.4:compile
[INFO]    +- org.webjars:x-editable-bootstrap:jar:1.5.1:compile
[INFO]    +- org.webjars:spin-js:jar:2.1.0:compile
[INFO]    +- org.webjars:animate.css:jar:3.3.0:compile
[INFO]    +- org.webjars.bower:summernote:jar:0.8.1:compile
[INFO]    +- org.apache.commons:commons-lang3:jar:3.4:compile
[INFO]    \- de.agilecoders.maven:maven-parent-config:jar:0.3.5:compile
```

**Comment by tmohme** [comment]
The corresponding output of `./gradlew  dependencies --configuration compile`:

>:dependencies
>
>------------------------------------------------------------
>Root project
>------------------------------------------------------------
>
>compile - Dependencies for source set 'main'.
>\--- de.agilecoders.wicket:wicket-bootstrap-extensions:0.10.11
>     +--- de.agilecoders.wicket:jquery-selectors:0.2.3
>     |    +--- com.google.guava:guava:18.0
>     |    +--- com.fasterxml.jackson.core:jackson-core:2.6.0
>     |    \--- com.fasterxml.jackson.core:jackson-databind:2.6.0
>     |         +--- com.fasterxml.jackson.core:jackson-annotations:2.6.0
>     |         \--- com.fasterxml.jackson.core:jackson-core:2.6.0
>     +--- de.agilecoders.wicket:wicket-bootstrap-core:0.10.11
>     |    +--- de.agilecoders.wicket:jquery-selectors:0.2.3 (*)
>     |    +--- de.agilecoders.wicket.webjars:wicket-webjars:0.5.5
>     |    |    +--- org.apache.wicket:wicket-core:7.4.0
>     |    |    |    +--- org.apache.commons:commons-collections4:4.1
>     |    |    |    +--- org.apache.wicket:wicket-request:7.4.0
>     |    |    |    |    +--- org.apache.wicket:wicket-util:7.4.0
>     |    |    |    |    |    +--- commons-fileupload:commons-fileupload:1.3.2
>     |    |    |    |    |    |    \--- commons-io:commons-io:2.2 -> 2.5
>     |    |    |    |    |    +--- commons-io:commons-io:2.5
>     |    |    |    |    |    \--- org.slf4j:slf4j-api:1.7.21
>     |    |    |    |    \--- org.slf4j:slf4j-api:1.7.21
>     |    |    |    +--- org.apache.wicket:wicket-util:7.4.0 (*)
>     |    |    |    \--- org.slf4j:slf4j-api:1.7.21
>     |    |    \--- org.slf4j:slf4j-api:1.7.21
>     |    +--- org.webjars:bootstrap:3.3.7-1
>     |    |    \--- org.webjars:jquery:1.11.1
>     |    +--- org.webjars:modernizr:2.8.3
>     |    +--- joda-time:joda-time:2.3
>     |    +--- org.apache.wicket:wicket-core:6.15.0 -> 7.4.0 (*)
>     |    +--- org.apache.wicket:wicket-extensions:6.15.0
>     |    |    +--- org.apache.wicket:wicket-core:6.15.0 -> 7.4.0 (*)
>     |    |    \--- org.slf4j:slf4j-api:1.6.4 -> 1.7.21
>     |    +--- org.apache.wicket:wicket-request:6.15.0 -> 7.4.0 (*)
>     |    +--- org.apache.wicket:wicket-util:6.15.0 -> 7.4.0 (*)
>     |    +--- org.slf4j:slf4j-api:1.7.7 -> 1.7.21
>     |    +--- org.apache.commons:commons-lang3:3.4
>     |    \--- de.agilecoders.maven:maven-parent-config:0.3.4
>     +--- org.apache.wicket:wicket-extensions:6.15.0 (*)
>     +--- org.apache.wicket:wicket-util:6.15.0 -> 7.4.0 (*)
>     +--- org.webjars:Eonasdan-bootstrap-datetimepicker:4.17.37-1
>     |    +--- org.webjars:bootstrap:3.1.1-2 -> 3.3.7-1 (*)
>     |    \--- org.webjars:momentjs:2.10.3 -> 2.14.1
>     +--- org.webjars:momentjs:2.14.1
>     +--- org.webjars:font-awesome:4.6.3
>     +--- com.google.guava:guava:14.0.1 -> 18.0
>     +--- com.google.javascript:closure-compiler:v20130603
>     |    +--- args4j:args4j:2.0.16
>     |    +--- com.google.guava:guava:14.0.1 -> 18.0
>     |    +--- com.google.protobuf:protobuf-java:2.4.1
>     |    +--- org.json:json:20090211
>     |    \--- com.google.code.findbugs:jsr305:1.3.9
>     +--- joda-time:joda-time:2.3
>     +--- org.webjars:jquerypp:1.0.1
>     +--- org.webjars:jquery-ui:1.11.4
>     |    \--- org.webjars:jquery:1.11.1
>     +--- org.webjars:typeaheadjs:0.10.4
>     +--- org.webjars:x-editable-bootstrap:1.5.1
>     |    \--- org.webjars:bootstrap:3.0.0 -> 3.3.7-1 (*)
>     +--- org.webjars:spin-js:2.1.0
>     +--- org.webjars:animate.css:3.3.0
>     +--- org.webjars.bower:summernote:0.8.1
>     +--- org.apache.commons:commons-lang3:3.4
>     \--- de.agilecoders.maven:maven-parent-config:0.3.4
>
>(*) - dependencies omitted (listed previously)

There are several other remarkable differences, e.g. :
* gradle lists `org.apache.wicket:wicket-extensions` as a dependency of `de.agilecoders.wicket:wicket-bootstrap-core`, maven doesn't
* de.agilecoders.maven:maven-parent-config:jar:0.3.5 vs. 0.3.4

I haven't investigated these other differences yet, so I can't tell if they might have the same or a different cause.

**Comment by tmohme** [comment]
In case my answer answers weren't clear enough:   
Yes, I have verified and can confirm that Maven resolves `wicket-extension` to 7.4.0 while Gradle resolves it to 6.15.0

**Comment by bmuschko** [comment]
Thanks for reporting back. Looks like a bug to me then. Are you interested in working on a fix?

**Comment by tmohme** [comment]
Generally yes. 
I'm just not sure that I can find enough spare time to dig through the code and *really* understand this core part of gradle in order to provide something helpful and not introduce new subtle problems.

**Comment by ljacomet** [comment]
Looks like Gradle 4.0 already resolved `wicket` to `7.4.0` like Maven, so closing
